types/lodash/common/array.d.ts

```diff
@@ -574,6 +574,7 @@ declare module "../index" {
          * @param array The array to query.
          * @return Returns the first element of array.
          */
+        head<T extends readonly unknown[]>(array: T): never[] extends T ? T[number] | undefined : T[0];
```

Comment:
Given this requires an overload, I think it may actually be better to not use a conditional type here. Rather, you can write:

```ts
head<T>(array: readonly [T, ...unknown[]]): T;
```

Which I think also works, and has the benefit of not making `T` refer to two entirely different things between the overloads. It also seems to work for non-const array literals passed in, since TS can recognize that it's being passed in an array literal that could be used as tuple.

[Playground Link](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEYD2A7AzgF3rGAuemMAligOYDaAugNwCwAUA6JLAgGYCuKYGRq8ACxBRgAHgAq8EAA8MIFMDTw4I1BACe8LgGsUSAO4oqAPgAUOKOvziAlPhQgAbiBhUps+YviSA-N-IoHAC2AEYulPAAPloKIGwkIMDw1uQADDRM4NBw8JzcvPxCIhJmFlbKwsBqmuJu0YEQEFExoPEOwHbezVytCcB0jPQA3gzwY4iomNjwALyClabkAIwANPAATGsAzJQ2A+PwAPSHBwB6PqPjyOhYIbPzIourG9sRUErXmHuXY8dnF-QDp8sGB7kVgItdvtxn9xucfhMbvAknNweYYDBvoCYSc4QCAL4MTIsHJ5Hh8FAPYDrEromCWfAqKooDTwcjiNYAOm5Oj0hiou2sA2Y2XYXHJhUqNPEpQxDIqqhZNTq8AaTWiPTifU6kg1sTaiQGDBG2LGwOmqKlTzWm3gOyxB1hY3hpsRUzulseyxtr2wH0mGAdOP+CPNoM9EKoQd+uOdAKBAeRYIWOGjR1j8Bd+KAA)

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70615/commits/a2f963771635c4d8f89a3d88e310678d7c8e1787

Thanks for the suggestion! I've changed conditional type to variadic tuple.